src/libtorch.cc

```diff
@@ -607,7 +726,19 @@ ModelInstanceState::ModelInstanceState(
   }
 
   THROW_IF_BACKEND_INSTANCE_ERROR(model_state->LoadModel(
-      ArtifactFilename(), device_, &model_path_, &torch_model_));
+      ArtifactFilename(), device_, &model_path_, Kind(), &torch_model_));
+
+  if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_MODEL) {
+#ifdef TRITON_ENABLE_GPU
+    // Create a CUDA stream for every availble device.
```

Comment:
Is there no way to capture what cuda devices are used by this model instance?
It's good that we are synchronizing against all these streams before capturing first_compute_infer_start_ event, otherwise the timestamps would have been skewed. You should document this implication of creating streams and events on all available devices. So, that in future we are careful when working with this code. 
For example, imagine if someone is capturing first_compute_output_start_ event on a device that isn't running the model.

Reply:
Added comment in this [commit](https://github.com/triton-inference-server/pytorch_backend/commit/65e8cb49481903f0ef33211fbbcc4576459e27d1). Not sure why it's not showing on the PR..